src/Symmetry/Anisotropy.jl

```diff
@@ -0,0 +1,535 @@
+# N-dimensional irreducible matrix representation of 𝔰𝔲(2).
+# TODO: Unify with David's code.
+function spin_operators(N)
+    s = (N-1)/2
+    a = 1:N-1
+    off = @. sqrt(2(s+1)*a - a*(a+1)) / 2
+
+    Sx = diagm(1 => off, -1 => off)
+    Sy = diagm(1 => -im*off, -1 => +im*off)
+    Sz = diagm((N-1)/2 .- (0:N-1))
+    return SVector{3, Matrix{ComplexF64}}(Sx, Sy, Sz)
+end
```

Comment:
As you note, a version of this function exists in `Interactions.jl`, where it is called [gen_spin_ops](https://github.com/SunnySuite/Sunny.jl/blob/17cf0b3be9fdae5bd6d3c281ac1439ac88bb353d/src/Interactions.jl#L233).